src/main/scala/regmapper/Annotation.scala

```diff
@@ -2,22 +2,196 @@
 
 package freechips.rocketchip.regmapper
 
+import chisel3.internal.InstanceId
+import chisel3.experimental.{ChiselAnnotation, RunFirrtlTransform, annotate}
+import firrtl.{CircuitForm, CircuitState, LowForm, Transform}
+import firrtl.annotations._
+import freechips.rocketchip.regmapper._
+import org.json4s.JsonAST.JValue
 import org.json4s.JsonDSL._
 import org.json4s.jackson.JsonMethods.{pretty, render}
+import org.json4s._
+import org.json4s.native.JsonMethods._
+
+
+/**
+  * This function is called by DescribedRegChiselAnnotation which creates a RegFieldDescAnnotation
+  * A backend Firrtl transform 'looks' for RegFieldDescAnnotation's attached to circuits and then processes
+  * the discovered RegFieldDescAnnotation's.
+   */
+case class RegFieldDescAnnotation(
+    target: Named,
+    desc: Seq[String]) extends SingleTargetAnnotation[Named] {
+  def duplicate(n: Named): RegFieldDescAnnotation = this.copy(target = n)
+}
+
+object RegFieldDescAnnotation {
+  def nameMap(state: CircuitState): Map[String,Seq[String]] = {
+    state.annotations.collect {
+      case RegFieldDescAnnotation(mname, cname) => mname.toString -> cname
+    }.toMap
+  }
+}
+
+
+case class DescribedRegChiselAnnotation(
```

Comment:
I think we chatted about this, but there is no need for a `DescribedReg` annotation. `DescribedReg` is just a convenience for the designer in coding up a `RegFieldDesc`. It is only once a `RegFieldDesc` is applied to a `RegField` that is passed into a call to `regmap()` that we care about it. At the point of the `DescribedReg` call it is meaningless.

Reply:
Not sure if you still have a pending change, but the DescribedReg annotation still exists in this version

Reply:
I added it as an initial proposal for @depappas-sifive to work from, but I think we've decided to go the route of annotating the entire map at once inside the call to `regmap`.

I think the only thing we're giving up is the opportunity to annotate the specific `Reg` component with its description (rather than annotation the enclosing `Module` with the entire map). Annotating the `Reg` component in particular would make it much easier to extract e.g. the final signal name. Sadly, not all information is known inside of the call to `DescribedReg` (like the base address). A combination approach might be preferable but we can try that in a follow-up PR.

Reply:
Yeah... but I expect there will be many more `RegFieldDesc` than there are `DescribedRegs`.

Reply:
removed

Reply:
Removed